src/zjs_callbacks.c

```diff
@@ -388,6 +388,53 @@ void print_callbacks(void)
 #define print_callbacks() do {} while (0)
 #endif
 
+#define MAX_ERROR_NAME_LENGTH       32
+#define MAX_ERROR_MESSAGE_LENGTH    128
+
+static void print_error_message(jerry_value_t error)
+{
+    uint32_t size;
+    char* message = NULL;
+    jerry_value_t name_val = zjs_get_property(error, "name");
+    if (!jerry_value_is_string(name_val)) {
+        ERR_PRINT("error did not have name\n");
+        // we should never get here.
+        return;
+    }
+    jerry_value_t msg_val = zjs_get_property(error, "message");
+    if (!jerry_value_is_string(msg_val)) {
+        ERR_PRINT("error did not have message\n");
+        // we should never get here.
+        return;
+    }
+
+    size = MAX_ERROR_NAME_LENGTH;
+    char name[size];
+
+    zjs_copy_jstring(name_val, name, &size);
+    if (!size) {
+        ERR_PRINT("name length is too long\n");
```

Comment:
Another return case where you're not freeing those other values. This also kind of encourages structuring your code with a smaller number of return points I guess. For better or worse. Wish we had smart pointers. :)